File: TMS/wwwroot/js/Teacher/Teacher.Model.js
/// <reference path="../Knockout.js" />
var TeacherModel = function (item) {
    var self = this;
    item = item || {};
    self.id = ko.observable(item.id || 0);
    self.firstName = ko.observable(item.firstName || '');
    self.lastName = ko.observable(item.lastName || '');
    self.address = ko.observable(item.address || '');
    self.gender = ko.observable(item.gender || '');
    self.dob = ko.observable(item.dob || '');
    self.qualification = ko.observable(item.qualification || '');
    self.salary = ko.observable(item.salary || '');
}

File: TMS/Models/TeacherModel.cs
using System.ComponentModel.DataAnnotations;

namespace TMS.Models
{
    public class TeacherModel
    {
        [Key]
        public int id { get; set; }
        [Required]
        [MaxLength(20)]
        public string firstName { get; set; }
        [Required]
        [MaxLength (20)]
        public string  lastName { get; set; }
        [Required]
        public string address { get; set; }
        [Required]
        public string gender { get; set; }  
        public DateTime dob { get; set; }
        [Required]
        public string qualification { get; set; }
        public double salary { get; set; }


    }
}
